closure/goog/dom/xml.js

```diff
@@ -39,6 +39,33 @@ goog.dom.xml.MAX_ELEMENT_DEPTH = 256; // Same default as MSXML6.
 
 
 /**
+ * Check for ActiveXObject support by the browser.
+ * @return {boolean} true if browser has ActiveXObject support.
+ * @private
+ */
+goog.dom.xml.hasActiveXObjectSupport_ = function() {
+  try {
+    // Due to lot of changes in IE 9, 10 & 11 behaviour, trying to create the
+    // ActiveXOjbect is a lot more reliable than testing for the existance of
+    // window.ActiveXObject
+    new ActiveXObject('MSXML2.DOMDocument');
```

Comment:
As ActiveX usage can be disabled using MSIE's security level, it is not reliable to only use goog.userAgent.IE to test for usage of ActiveXObject... In fact, I do think that trying to instantiate the ActiveXObject is the only reliable way of testing for ActiveX support.

For the global scope issue, is this a better (not perfect) solution ?
```javascript
if (goog.userAgent.IE) {
  /**
   * Check for ActiveXObject support by the browser.
   * @return {boolean} true if browser has ActiveXObject support.
   * @private
   */
  goog.dom.xml.hasActiveXObjectSupport_ = function() {
    [...]
  };

  /**
   * True if browser has ActiveXObject support.
   * Possible override if this test become wrong in coming IE versions.
   * @type {boolean}
   */
  goog.dom.xml.ACTIVEX_SUPPORT =
    goog.dom.xml.hasActiveXObjectSupport_();

} else {
  /**
   * True if browser has ActiveXObject support.
   * Possible override if this test become wrong in coming IE versions.
   * @type {boolean}
   */
  goog.dom.xml.ACTIVEX_SUPPORT = false;
}
```